src/main/java/com/google/sps/servlets/EventFreeTimeServlet.java

```diff
@@ -0,0 +1,198 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.batch.BatchRequest;
+import com.google.api.client.googleapis.batch.json.JsonBatchCallback;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonError;
+import com.google.api.client.googleapis.json.GoogleJsonErrorContainer;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.FreeBusyCalendar;
+import com.google.api.services.calendar.model.FreeBusyRequest;
+import com.google.api.services.calendar.model.FreeBusyRequestItem;
+import com.google.api.services.calendar.model.FreeBusyResponse;
+import com.google.api.services.calendar.model.TimePeriod;
+import com.google.appengine.api.ThreadManager;
+import com.google.appengine.api.users.UserService;
+import com.google.gson.Gson;
+import com.google.sps.data.DateTimeRange;
+import com.google.sps.data.EventStorage;
+import com.google.sps.data.Time;
+
+@WebServlet("/freetimes")
+public class EventFreeTimeServlet extends HttpServlet {
+  AuthorizationCodeFlow flow;
+  EventStorage eventStorageObject;
+  UserService userService;
+
+  @Inject
+  EventFreeTimeServlet(AuthorizationCodeFlow flow, EventStorage eventStorageObject, UserService userService) {
+    this.flow = flow;
+    this.eventStorageObject = eventStorageObject;
+    this.userService = userService;
+  }
+
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
```

Comment:
I would extract the get busy times logic into a separate class to make it also testable in the end because it calls the calendar API.